src/main/java/de/braintags/vertx/util/file/JsonReadStream.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * #%L
+ * Vert.x utilities from Braintags
+ * %%
+ * Copyright (C) 2017 Braintags GmbH
+ * %%
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * #L%
+ */
+package de.braintags.vertx.util.file;
+
+import java.util.Iterator;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.annotation.Nullable;
+
+import de.braintags.vertx.util.ExceptionUtil;
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.Json;
+import io.vertx.core.streams.Pump;
+import io.vertx.core.streams.ReadStream;
+
+/**
+ * A {@link ReadStream} which takes a List of Objects and streams them as Json. Can be used as source for {@link Pump}
+ * 
+ * @author Michael Remme
+ * @param <T>
+ *          the type of the instances to be streamed
+ */
+public class JsonReadStream<T> implements ReadStream<Buffer> {
+  private static final io.vertx.core.logging.Logger LOGGER = io.vertx.core.logging.LoggerFactory
+      .getLogger(JsonReadStream.class);
+
+  private final AtomicBoolean paused = new AtomicBoolean(true);
+  private final AtomicBoolean ended = new AtomicBoolean(false);
+  private Handler<Throwable> exceptionHandler = new DefaultExceptionHandler();
+  private Handler<Buffer> contentHandler;
+  private Handler<Void> endHandler;
+  private final Iterator<T> instances;
+  private boolean pretty = false;
+  private boolean array = true;
+  private final AtomicBoolean firstElementWritten = new AtomicBoolean(false);
+
+  /**
+   * the same as this( instances, false, true )
+   * 
+   * @param instances
+   */
+  public JsonReadStream(final Iterable<T> instances) {
+    this(instances, false, true);
+  }
+
+  /**
+   * 
+   * @param instances
+   *          the instances to be streamed
+   * @param pretty
+   *          wether output shall be formatted json
+   * @param array
+   *          wether output shall be array. if false, then each entity is written into one line as json
+   */
+  public JsonReadStream(final Iterable<T> instances, final boolean pretty, final boolean array) {
+    this.instances = instances.iterator();
+    this.pretty = pretty;
+    this.array = array;
+  }
+
+  @Override
+  public ReadStream<Buffer> exceptionHandler(final Handler<Throwable> handler) {
+    this.exceptionHandler = handler;
+    return this;
+  }
+
+  @Override
+  public ReadStream<Buffer> handler(@Nullable final Handler<Buffer> handler) {
+    this.contentHandler = handler;
+    resume();
+    return this;
+  }
+
+  private void next() {
+    if (!paused.get()) {
+      if (instances.hasNext()) {
+        contentHandler.handle(createContent(instances.next()));
```

Comment:
possible NPE, if content handler is not set

Reply:
added dedicated NPE into resume method when no content handler is set